op-node/rollup/driver/sequencer.go

```diff
@@ -106,7 +106,9 @@ func (d *Sequencer) CreateNewBlock(ctx context.Context, l2Head eth.L2BlockRef, l
 	// TODO: allowing to breathe when remaining time is in the negative is very generous,
 	//  we can reduce this if the block building timing gets better with PR 3818
 	d.log.Debug("using remaining time for better block production", "remaining_time", remaining)
-	time.Sleep(500 * time.Millisecond)
+	if remaining < time.Minute {
+		time.Sleep(2 * time.Second * time.Duration(d.config.BlockTime))
```

Comment:
```suggestion
	// If there is more than 2 blocks worth of time remaining, then something is wrong with scheduling, abort.
	if remaining > 2 * time.Second * time.Duration(d.config.BlockTime) {
		return nil,  fmt.Errorf("block is being created too far in advance: we have %s remaining time", remaining)
	} else if remaining > 100 * time.Millisecond {
		// If we are not too close to the head time, then we are not in a rush to produce the block and we can take a breathe to include more transactions.
		time.Sleep(remaining - 90 * time.Millesecond) // leave 0.09s for sealing the block, and do not sleep(0)
```

Something like this works better. We should never wait for 2 seconds when we e.g. only have 40 ms left before the block timestamp, and we should not wait more than the remaining time (minus some margin for wrapping up the work).